heat/core/tests/test_arithmetics.py

```diff
@@ -98,44 +109,173 @@ def test_add(self):
         with self.assertRaises(TypeError):
             self.a_tensor + "s"
 
+    def test_add_(self):
+        # Copies of class variables for the in-place operations
+        a_scalar = 2.0
+        a_vector = self.a_vector.copy()
+        another_vector = self.another_vector.copy()
+        a_tensor = a_tensor_double = self.a_tensor.copy()
+        another_tensor = self.another_tensor.copy()
+        a_split_tensor = a_split_tensor_double = self.a_split_tensor.copy()
+        an_int_scalar = self.an_int_scalar
+        erroneous_type = self.erroneous_type
+        a_string = self.a_string
+
+        result = ht.array([[3.0, 4.0], [5.0, 6.0]])
+
+        # We identify the underlying PyTorch objects to check whether operations are really in-place
+        underlying_torch_tensor = a_tensor.larray
+        underlying_split_torch_tensor = a_split_tensor.larray
+
+        # Check for every possible combination of inputs whether the right solution is computed and
+        # saved in the right place and whether the second input stays unchanged. After every tested
+        # computation, we reset changed variables.
+        self.assertTrue(ht.equal(a_tensor.__iadd__(a_scalar), result))  # test result
+        self.assertTrue(ht.equal(a_tensor, result))  # test result in-place
+        self.assertIs(a_tensor, a_tensor_double)  # test in-place
+        self.assertIs(a_tensor.larray, underlying_torch_tensor)  # test in-place
+        self.assertTrue(ht.equal(a_scalar, ht.float32(2.0)))  # test if other input is unchanged
+        a_tensor = a_tensor_double = ht.array([[1.0, 2.0], [3.0, 4.0]])  # reset
+        underlying_torch_tensor = a_tensor.larray  # reset
```

Comment:
```suggestion
        underlying_torch_tensor.copy_(self.a_tensor.larray)  # reset
```
Regarding the last point: Now, you can write the reset in a single line and save some space :grinning: